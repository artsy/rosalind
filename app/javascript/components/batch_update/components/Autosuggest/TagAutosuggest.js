import React from 'react'
import PropTypes from 'prop-types'
import GenericAutosuggest from './GenericAutosuggest'
import { matchTags } from 'lib/rosalind-api'
import { getSuggestionValue, renderSuggestion } from './helpers'

function TagAutosuggest(props) {
  return (
    <GenericAutosuggest
      id="tag-autosuggest"
      placeholder={props.placeholder}
      fetchSuggestions={matchTags}
      getSuggestionValue={getSuggestionValue}
      renderSuggestion={renderSuggestion}
      selectSuggestion={tag => {
        props.onSelectTag(tag)
      }}
    />
  )
}

TagAutosuggest.propTypes = {
  onSelectTag: PropTypes.func.isRequired,
}

TagAutosuggest.defaultProps = {
  placeholder: 'Select a tag',
}

export { TagAutosuggest }
