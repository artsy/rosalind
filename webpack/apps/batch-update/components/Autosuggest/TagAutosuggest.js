import React from 'react'
import GenericAutosuggest from './GenericAutosuggest'
import { matchTags } from 'lib/rosalind-api'
import { getSuggestionValue, renderSuggestion } from './helpers'

function TagAutosuggest (props) {
  return (
    <GenericAutosuggest
      id='tag-autosuggest'
      placeholder={props.placeholder}
      fetchSuggestions={matchTags}
      getSuggestionValue={getSuggestionValue}
      renderSuggestion={renderSuggestion}
      selectSuggestion={tag => { props.onSelectTag(tag) }}
      />
  )
}

TagAutosuggest.propTypes = {
  onSelectTag: React.PropTypes.func.isRequired
}

TagAutosuggest.defaultProps = {
  placeholder: 'Select a tag'
}

export { TagAutosuggest }
