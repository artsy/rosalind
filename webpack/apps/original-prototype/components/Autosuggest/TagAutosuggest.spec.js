import React from 'react'
import renderer from 'react-test-renderer'
import { TagAutosuggest } from './TagAutosuggest'

const mockHandler = jest.fn()

describe('TagAutosuggest', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<TagAutosuggest onSelectTag={mockHandler} />)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

