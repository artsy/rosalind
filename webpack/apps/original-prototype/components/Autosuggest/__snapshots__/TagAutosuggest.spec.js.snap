exports[`TagAutosuggest renders correctly 1`] = `
<div
  className="GenericAutosuggest">
  <div
    className="react-autosuggest__container">
    <input
      aria-activedescendant={null}
      aria-autocomplete="list"
      aria-expanded={false}
      aria-haspopup={false}
      aria-owns="react-autowhatever-tag-autosuggest"
      autoComplete="off"
      className="react-autosuggest__input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Select a tag"
      role="combobox"
      type="text"
      value="" />
    <div
      className="react-autosuggest__suggestions-container"
      id="react-autowhatever-tag-autosuggest" />
  </div>
</div>
`;
